sympy/solvers/decompogen.py

```diff
@@ -37,14 +37,24 @@ def decompogen(f, symbol):
     if isinstance(f, (Function, Pow)):
         if f.args[0] == symbol:
             return [f]
+
+        if f.is_Pow:
+            base, expo = f.args
+            if expo.has(symbol):
+                return [f]
+
         result += [f.subs(f.args[0], symbol)] + decompogen(f.args[0], symbol)
```

Comment:
I think a better way here would be to make it handle multivariate functions, and then treat Pow as a multivariate function. In other words, one should decompose all the args, not just args[0].

Reply:
@asmeurer I didn't quite get the idea. Like if `f=2**x`, f.args are `(2, x)`. How will these be decomposed?

Reply:
In that case it isn't decomposable, but consider `2**(x + 1)` or `(x + 1)**2`. In both cases you can decompose `x + 1`.

Reply:
@asmeurer sorry for the late reply.
If I try to decompose all the args the anomaly would still remain. Like if `f=2**(x+1)`, f.args would be `(2, x+1)`, and again we are trying to decompose `2`, which would result in the error mentioned at first.

Another way can be is making a separate case for exponents, because for such equations only the argument to be decomposed lies in `f.args[1]` rather than `f.args[0]`.

Reply:
I have implemented as I said above.
Is it good enough?